### DISCUSSION TITLE
[FEATURE REQUEST] Get the past sponsorships.

### DISCUSSION THREAD

**Discussion by dohooo** [community/community#38393] [Other Feature Feedback, Questions, & Ideas]
Re. [Displaying Past Sponsorships on Sponsors Profile Page](https://github.com/community/community/discussions/37234).

Uhhh, What's the GraphQL API endpoint?

**Comment by jfullstackdev** [comment]
if you are asking what is GraphQL, check this out 

https://graphql.org/

it's the latest update from the REST API, check this out

https://github.blog/2016-09-14-the-github-graphql-api/

**Reply by dohooo** [reply]
Thanks for your answer, but I wanna got a field that can get the past sponsorships. I've searched it from documentation of github graphql, but I can't.

**Comment by cheshire137** [comment]
Hi there! I'm a dev on the Sponsors team. We're adding the ability to query for past sponsorships to the GraphQL API. It's a work in progress, but the first chunk just became available: there's now an `activeOnly` argument for the `Sponsorable.sponsorshipsAsSponsor` connection. For example:

```graphql
{
  viewer {
    login
    sponsorshipsAsSponsor(first: 100, activeOnly: false) {
      totalCount
      nodes {
        sponsorable {
          ... on User {
            login
          }
          ... on Organization {
            login
          }
        }
      }
    }
  }
}
```

When I run this on https://docs.github.com/en/graphql/overview/explorer, I get 20 results because my past sponsorships are included, versus the 11 I get when I either a) omit `activeOnly` or b) pass activeOnly=true.

Hopefully this helps! I'll post on here again if more fields or arguments are added to the API involving past sponsorships.

**Reply by dohooo** [reply]
Thanks for your work!

**Reply by cheshire137** [reply]
Okay, got another change up for the GraphQL API! Now the `Sponsorable.sponsorshipsAsMaintainer` connection also accepts an optional `activeOnly` argument. For example:

```graphql
query {
  organization(login:"someMaintainer") {
    pastAndCurrent: sponsorshipsAsMaintainer(activeOnly: false) {
      totalCount
    }
    currentOnly: sponsorshipsAsMaintainer(activeOnly: true) {
      totalCount
    }
  }
}
```

**Reply by dohooo** [reply]
Is there any approach that can only get past sponsorships data?

**Reply by cheshire137** [reply]
Not currently, no! Is that something you're interested in?

**Reply by dohooo** [reply]
Emm，Or are there any fields that distinguish past sponsors?

**Reply by dohooo** [reply]
Because I'm developing a tool that can generate sponsorship coverage and the form of display different between current and past, so I need to distinguish them. I need to fetch two times and process them with a filter if I can not get two kinds of data directly.

**Reply by cheshire137** [reply]
> are there any fields that distinguish past sponsors?

Not yet, but I'm working on a `Sponsorship.isActive` field that I expect to ship soon.

**Reply by cheshire137** [reply]
The `Sponsorship.isActive` field is now available! For example:

![a GraphQL query showing an inactive sponsorship and an active sponsorship](https://user-images.githubusercontent.com/82317/211114187-54b8f795-61cf-44c1-a377-04554e99cf54.png)

-- https://docs.github.com/en/graphql/overview/explorer

```graphql
query {
  mikemcquaid:user(login:"mikemcquaid") {
    sponsorshipForViewerAsSponsor(activeOnly: false) {
      isActive
      sponsorEntity {
        ... on User { login }
      }
    }
  }
  jonschlinkert:user(login:"jonschlinkert") {
    sponsorshipForViewerAsSponsor(activeOnly: false) {
      isActive
      sponsorEntity {
        ... on User { login }
      }
    }
  }
}
```

You can see on my profile page, https://github.com/cheshire137?tab=sponsoring, that I'm currently sponsoring `jonschlinkert` (`isActive` came back as true in the API response above) and that I had previously sponsored `mikemcquaid` (`isActive` came back as false in the API response above).

**Reply by dohooo** [reply]
C O O L !

**Reply by ralphjsmit** [reply]
Hello @cheshire137, I just came across your posts here about the available fields on the GraphQL API about sponsors :) I'm doing academic research and the `isActive` field is great – however I would love to find a way to see when a sponsorship ended? Is it possible for inactive sponsorships to get the end-date of the sponsorship? Or would there be another better way? Thank you very much!
